hw13/task2/proxy_txt_reader.py

```diff
@@ -0,0 +1,37 @@
+from hw13.task2.abstract_reader import Reader
+from hw13.task2.txt_reader import TxtReader
+
+
+class TxtProxyReader(Reader):
+
+    def __init__(self, txt_path):
+        self.__result = ''
+        self.__is_actual = False
+        self.__reader = TxtReader(txt_path)
+        self.__text = ''
+        self.__writer = TxtReader(txt_path)
+
+    def read_file(self):
+        if self.__is_actual:
+            return self.__result
+        else:
+            self.__result = self.__reader.read_file()
+            self.__is_actual = True
+            print(self.__result)
+            return self.__result
+
+    def write_file(self, text):
+        self.__text = text
+        self.__writer.write_file(text)
+        self.__is_actual = False
```

Comment:
and where is proxy? Think about moment when there is no need for writing to file